src/util/http/HttpClient.cpp

```diff
@@ -204,20 +207,60 @@ HttpOrHttpsResponse sendHttpOrHttpsRequest(
     const ad_utility::httpUtils::Url& url,
     ad_utility::SharedCancellationHandle handle,
     const boost::beast::http::verb& method, std::string_view requestData,
-    std::string_view contentTypeHeader, std::string_view acceptHeader) {
-  auto sendRequest = [&](auto ti) -> HttpOrHttpsResponse {
+    std::string_view contentTypeHeader, std::string_view acceptHeader,
+    size_t maxRedirects) {
+  auto sendRequest = [&](const Url& currentUrl,
+                         auto ti) -> HttpOrHttpsResponse {
     using Client = typename decltype(ti)::type;
-    auto client = std::make_unique<Client>(url.host(), url.port());
-    return Client::sendRequest(std::move(client), method, url.host(),
-                               url.target(), std::move(handle), requestData,
+    auto client =
+        std::make_unique<Client>(currentUrl.host(), currentUrl.port());
```

Comment:
I think it would be better to only instantiate the client once, not once for every redirect step. This way the underlying TCP socket gets re-used.

Reply:
@RobinTF Redirects can (and often will) change the host and port, and then you need a new connection. I wouldn't bother optimizing this only for the case where the redirects are happening on the same host:port. OK?